### PR TITLE
fix error when parsing diff that deleted empty files.

### PR DESCRIPTION
**Pull request by shogo82148**
When I delete an empty file, the reviewdog doesn't work as intended.

Reproduction procedure:

``` console
$ git init
$ touch .empty-file
$ git add .empty-file
$ git commit -m 'init commit'
[master (root-commit) 8a2e61c] init commit
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 .empty-file
$ git checkout -b new-branch
Switched to a new branch 'new-branch'
$ git rm .empty-file
rm '.empty-file'
$ echo -e "package main\nvar FooBar int" > main.go
$ git add main.go
$ git commit -m 'add main.go'
[new-branch a6566e8] add main.go
 2 files changed, 2 insertions(+)
 delete mode 100644 .empty-file
 create mode 100644 main.go
$ golint | reviewdog -f=golint -diff="git diff master"
```

I want the following result, but get nothing.

``` plain
main.go:2:5: exported var FooBar should have comment or be unexported
```

The result of `git diff master` is here.

``` diff
diff --git a/.empty-file b/.empty-file
deleted file mode 100644
index e69de29..0000000
diff --git a/main.go b/main.go
new file mode 100644
index 0000000..4af8c68
--- /dev/null
+++ b/main.go
@@ -0,0 +1,2 @@
+package main
+var FooBar int
```

### CODE REVIEW

**Comment by shogo82148** [comment]
I see.
Please wait for a little.

**Comment by shogo82148** [comment]
Done.
I add a new test case. It fails on master branch.

``` console
$ go test ./...
ok  	github.com/haya14busa/reviewdog	0.158s
ok  	github.com/haya14busa/reviewdog/cmd/reviewdog	1.160s
--- FAIL: TestParseMultiFile (0.01s)
	parse_test.go:22: testdata/empty.diff
	parse_test.go:22: testdata/empty_deleted.diff
	parse_test.go:22: testdata/empty_new.diff
	parse_test.go:22: testdata/empty_space.diff
	parse_test.go:22: testdata/golint.diff
	parse_test.go:22: testdata/newline_and_empty_deleted.diff
	parse_test.go:55: want &diff.FileDiff{PathOld:"", PathNew:"", TimeOld:"", TimeNew:"", Hunks:[]*diff.Hunk{}, Extended:[]string{"diff --git a/empty.txt b/empty.txt", "deleted file mode 100644", "index e69de29..0000000"}}, got (*diff.FileDiff)(nil)
	parse_test.go:55: want &diff.FileDiff{PathOld:"a/golint.old.go", PathNew:"b/golint.new.go", TimeOld:"", TimeNew:"", Hunks:[]*diff.Hunk{(*diff.Hunk)(0xc420014820)}, Extended:[]string{"diff --git a/golint.old.go b/golint.new.go", "index 34cacb9..a727dd3 100644"}}, got (*diff.FileDiff)(nil)
	parse_test.go:22: testdata/nonewline.diff
	parse_test.go:22: testdata/nonewline2.diff
	parse_test.go:22: testdata/nonewline3.diff
	parse_test.go:22: testdata/sample.diff
	parse_test.go:22: testdata/sample.git.diff
FAIL
FAIL	github.com/haya14busa/reviewdog/diff	0.024s
ok  	github.com/haya14busa/reviewdog/project	0.094s
```

File: diff/parse.go [hunk 1]
```diff
@@ -103,7 +103,15 @@ func (p *fileParser) parseHunks() ([]*Hunk, error) {
 		return nil, ErrNoHunks
 	}
 	if !bytes.HasPrefix(b, []byte(tokenStartHunk)) {
-		return nil, ErrNoHunks
+		b, err := p.r.Peek(len(tokenDiffGit))
+		if err != nil {
+			return nil, ErrNoHunks
+		}
+		if !bytes.HasPrefix(b, []byte(tokenDiffGit)) {
```
Comment (haya14busa) [inline diff/parse.go]: Can you invert this condition?

This is special case for "git diff", not general unified diff. something like this.

```go
if bytes.HasPrefix(b, []byte(tokenDiffGit)) {
  // git diff may contain a file diff with empty hunks.
  // e.g. delete an empty file.
  return []*Hunk{}, nil
}
```

**Comment by haya14busa** [comment]
Huge thanks!!! :+1:

**Pull request merged**